kernel/memory/src/paging/mapper.rs

```diff
@@ -281,24 +319,70 @@ impl Mapper {
+                    } 
+        
+                    p1[page.p1_index()].set_entry(af.as_allocated_frame(), actual_flags);
+                    core::mem::forget(af); // we currently forget frames allocated here since we don't yet have a way to track them.
+                }
+            }
+            PAGE_2MB_SIZE => {
+                for page in pages.range_2mb().iter().step_by(512) {
+                    // This is just one frame for now, as you can only use set_entry on a single frame
+                    //let af = frame_allocator::allocate_frames(512).ok_or("map_allocated_pages(): couldn't allocate new frame, out of memory")?;
+                    let af = frame_allocator::allocate_frames(1).ok_or("map_allocated_pages(): couldn't allocate new frame, out of memory")?;
+
+                    let p3 = self.p4_mut().next_table_create(page.p4_index(), higher_level_flags);
+                    let p2 = p3.next_table_create(page.p3_index(), higher_level_flags);
+                    trace!("Attempting to map to level 2 PTE: {:?}", p2[page.p2_index()].flags());
+
+                    if !p2[page.p2_index()].is_unused() {
+                        error!("map_allocated_pages(): page {:#X} -> frame {:#X}, page was already in use!",
+                            page.start_address(), af.start_address()
+                        );
+                        return Err("map_allocated_pages(): page was already in use");
+                    }
+        
+                    p2[page.p2_index()].set_entry(af.as_allocated_frame(), actual_flags);
+                    core::mem::forget(af); // we currently forget frames allocated here since we don't yet have a way to track them.
+                }
+            }
+            PAGE_1GB_SIZE => {
+                for page in pages.range_1gb() {
+                    //let af = frame_allocator::allocate_frames(262114).ok_or("map_allocated_pages(): couldn't allocate new frame, out of memory")?;
+                    let af = frame_allocator::allocate_frames(1).ok_or("map_allocated_pages(): couldn't allocate new frame, out of memory")?;
+                    let p3 = self.p4_mut().next_table_create(page.p4_index(), higher_level_flags);
+                    trace!("Attempting to map to PTE: {:?}", p3[page.p3_index()].flags());
+
+                    if !p3[page.p1_index()].is_unused() {
+                        error!("map_allocated_pages(): page {:#X} -> frame {:#X}, page was already in use!",
+                            page.start_address(), af.start_address()
+                        );
+                        return Err("map_allocated_pages(): page was already in use");
+                    } 
+        
+                    p3[page.p3_index()].set_entry(af.as_allocated_frame(), actual_flags);
+                    core::mem::forget(af); // we currently forget frames allocated here since we don't yet have a way to track them.
+                }
+            }
```

Comment:
since these aren't yet implementable due to lack of support in the frame allocator, we shouldn't allow it to proceed. Instead, do this:

```suggestion
                todo!("Mapping 2MiB huge pages to randomly-allocated huge frames is not yet supported")
            }
            PAGE_1GB_SIZE => {
                todo!("Mapping 1GiB huge pages to randomly-allocated huge frames is not yet supported")
            }
```